src/ledgerSigner/index.js

```diff
@@ -245,6 +295,234 @@ class LedgerNanoS {
+    const btcLedger = await this.getBtcLedger()
+    const changeAddressPath = `${baseBtcPath}/0'/1/${changeIndex}`
+
+    let associatedKeysets = []
+    let finalInputs = []
+    let inputValueTotal = 0
+    for (let i = 0; i < inputs.length; i++) {
+      const utxo = inputs[i]
+      const utxoDetails = await this.getUtxoDetails(utxo.txHash)
+
+      const txObj = btcLedger.splitTransaction(utxoDetails, true)
+      const input = [txObj, utxo.outputIndex]
+      finalInputs.push(input)
+      associatedKeysets.push(utxo.keyPath)
+      inputValueTotal += utxo.value
+    }
+    let outputs = []
+    let amountBuffer = Buffer.alloc(8, 0)
+    amountBuffer.writeUIntLE(amount, 0, 8)
+    const txOutput = {
+      amount: amountBuffer,
+      script: address.toOutputScript(to, networks.testnet)
+    }
+    outputs.push(txOutput)
+    const change = inputValueTotal - amount - fee // 138 bytes for 1 input, 64 bytes per additional input
+
+    let changeBuffer = Buffer.alloc(8, 0)
+    changeBuffer.writeUIntLE(change, 0, 8)
+    const changeAddress = (await btcLedger.getWalletPublicKey(changeAddressPath, false, true)).bitcoinAddress
+    const changeOutput = {
+      amount: changeBuffer,
+      script: address.toOutputScript(changeAddress, networks.testnet)
+    }
+    outputs.push(changeOutput)
+
+    const outputScriptHex = btcLedger.serializeTransactionOutputs({ outputs: outputs }).toString('hex')
+    const signedTxRaw = await btcLedger.createPaymentTransactionNew(
+      finalInputs,
+      associatedKeysets,
+      changeAddressPath,
+      outputScriptHex,
+      undefined,
+      undefined,
+      true
+    )
+
+    return signedTxRaw
+  }
+
+  broadcastBtcRawTx = async (txRaw) => {
+    const rv = await axios.post(
+      `${ledgerApiUrl}/transactions/send`,
+      { tx: txRaw })
+    return rv.data.result
+  }
+
+  getUtxosFromTxs = (txs, address) => {
+    let utxos = []
+    let spent = {}
+    txs.forEach(tx => {
+      tx.inputs.forEach(input => {
+        if (input.address === address) {
+          if (!spent[input.output_hash]) {
+            spent[input.output_hash] = {}
+          }
+          spent[input.output_hash][input.output_index] = true
+        }
+      })
+    })
+    txs.forEach(tx => {
+      tx.outputs.forEach(output => {
+        if (output.address === address) {
+          if (!spent[tx.hash]) {
+            spent[tx.hash] = {}
+          }
+          if (!spent[tx.hash][output.output_index]) {
+            utxos.push({
+              txHash: tx.hash,
+              outputIndex: output.output_index,
+              value: output.value
+            })
+          }
+        }
+      })
+    })
+
+    return utxos
+  }
+
+  syncBtcAccountInfo = async (accountIndex) => {
+    const btcLedger = await this.getBtcLedger()
+    let i = 0
+    let totalBalance = 0
+    let addresses = []
+    let gap = 0
+    let utxos = []
+    let changeIndex = 0
+    let addressIndex = 0
+    while (gap < 5) {
```

Comment:
why using gap=5?

Reply:
also, looks like internal and external address discovery are in the same loop, shouldn't they be in separate loops?